packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/label/DiagramLabel.tsx

```diff
@@ -0,0 +1,52 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import { Label } from './Label';
+import React from 'react';
+import { DiagramLabelProps } from './DiagramLabel.types';
+import { LabelIcon } from './LabelIcon';
+import { LabelText } from './LabelText';
+
+const labelStyle = (
+  faded: Boolean,
+  hasIcon: boolean,
+  borderStyle: React.CSSProperties | undefined
+): React.CSSProperties => {
+  const labelStyle: React.CSSProperties = {
+    opacity: faded ? '0.4' : '',
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+    justifyContent: borderStyle?.justifyContent ?? 'flex-start',
```

Comment:
If someone needs the label at the start of some div, they can just move the whole `DiagramLabel` component. This option is not necessary.